src/base.js

```diff
@@ -216,8 +214,15 @@ export default class KintoClientBase {
       return this.serverInfo;
     }
     const path = this.remote + endpoint("root");
-    const reqOptions = this._getRequestOptions(options);
-    const { json } = await this.http.request(path, reqOptions);
+    const { json } = await this.http.request(
+      path,
+      {
+        headers: this._getHeaders(options),
+      },
+      {
+        retry: this._getRetry(options),
+      }
```

Comment:
Note that you can slightly influence the formatter, eg. if you write the object on a single line it will preserve that style. And yeah I wish it were fully idempotent.